### ISSUE TITLE
VAE - Training

### ISSUE THREAD

**Issue by palmex** [OpenMotionLab/MotionGPT#96]
I'm having a hard time recreating your results. I'm trying to retrain the VAE from scratch: LR 1e-4, Adam, 512 embedding size. The validation error seems to be leveling off, and I don't think brute forcing to 150K epochs would solve this issue. 

I'm only training on AMASS because I have not been able to get in contact with the author's of the HumanAct12 dataset. Would it be possible to a) share your loss function curves and b) get access to the HumanAct12 data? 

![Screenshot 2024-05-18 at 9 27 04 AM](https://github.com/OpenMotionLab/MotionGPT/assets/74429479/36e00c49-86a6-4bfc-9c3b-6b46764ef65c)

**Comment by palmex** [comment]
Need to turn debug mode off

**Comment by aixiaodewugege** [comment]
I meet the same problem. Could you please share the correct loss curve?

**Comment by palmex** [comment]
My problem ended up being that there is a debug flag which the author's used for iterating quickly. The debug mode only loads ~100 examples. Also, the batch size needs to be increased to 256 for VAE. Make sure you are using stage1 config. 

In the dataloader, there is also a caching function which checks for the existence of a ./tmp folder and loads a .pkl file instead of loading in the entire dataset again.  

![Screenshot 2024-12-10 at 11 16 36 AM](https://github.com/user-attachments/assets/13ba44e1-fe70-475b-8501-30eca43a4594)

**Comment by aixiaodewugege** [comment]
Thanks! It really helps! But my commit loss looks strange. Does yours look the same?
![image](https://github.com/user-attachments/assets/8af03772-a212-4d96-8dd6-af834abf0176)

**Comment by palmex** [comment]
Yes

**Comment by ChenFengYe** [comment]
@palmex @aixiaodewugege I am so sorry about your troubles caused by "this debug flag". If you have any other questions, feel free to ask. I'm more than happy to help.

**Comment by aixiaodewugege** [comment]
> @palmex @aixiaodewugege I am so sorry about your troubles caused by "this debug flag". If you have any other questions, feel free to ask. I'm more than happy to help.

"Hello, I'm a newcomer to VQ-VAEs and am currently working on training a face motion VQ-VAE using your code. However, I've noticed that the validation loss curve appears to be indicating overfitting. Could you please offer some advice to address this issue?"
![image](https://github.com/user-attachments/assets/d1d28447-609c-4101-a34b-a7d820fc3138)
 
![image](https://github.com/user-attachments/assets/cee9ec93-50c8-493f-a9a0-7d0d92634ae5)

**Comment by ChenFengYe** [comment]
@aixiaodewugege Based on the loss-feature/train and loss-feature/val curves, there is clear overfitting. The commit loss reflects the efficiency of the codebook, which seems somewhat correct. 

One question could help: how does this feature differ from the parameters of FLAME? 
A general solution to address this problem is to apply data augmentation and use dropout with masking on both input features and loss during training.

**Comment by aixiaodewugege** [comment]
@ChenFengYe
Thank you for your reply! It was really helpful. The FLAME model represents the vertex loss. After adding dropout, the results look better, but I still notice a gap between the training and validation loss. How can I minimize this gap?

Additionally, could you clarify what you mean by data augmentation?

**Comment by ChenFengYe** [comment]
@aixiaodewugege Data augmentation could help to minimize this gap. The specific process could be, for example,  RTS (rotation, translation, scale) on training data (like vertex, XYZ). You can augment your data by following your target tasks.
